lib/irc/formatting.js

```diff
@@ -164,103 +214,67 @@ module.exports.htmlToIrc = function(html) {
 };
 
 module.exports.ircToHtml = function(text) {
-    text = escapeHtmlChars(text);
-
-    // Color codes look like 003xx or 003xx,yy (xx=foreground, yy=background)
-    // We want to match until it hits another color (starting 003) or hits the
-    // reset code 000f. That is the 'end tag', represented as a non-greedy match
-    // for the reset code, or any character until 003.
-    //
-    //                  Foreground          Background      End 'tag'
-    //                 ______|_______       ____|____    _______|________
-    //                /              \     /         \  /                \
-    var colorRegex = /(\003[0-9]{1,2})[,]?([0-9]{1,2})?(.*?\u000f|[^\003]+)/;
-    var groups;
-    if (colorRegex.test(text)) {
-        groups = colorRegex.exec(text);
-        while (groups) {
-            // ignore bg for now (groups[2])
-
-            // this text includes the reset code so the code can be applied for
-            // each formatting code. We'll strip it later on.
-            var coloredText = groups[3];
-            var fontColor = colorCodesToHtmlNames[groups[1]];
-            if (fontColor) {
-                text = text.replace(
-                    groups[0],
-                    function() { // eslint-disable-line no-loop-func
-                        return '<font color="' + fontColor + '">' +
-                        coloredText +
-                        '</font>'
-                    }
-                );
-            }
-            else {
-                // unknown font colour
-                text = text.replace(groups[0], coloredText);
-            }
-            groups = colorRegex.exec(text);
-        }
-    }
+    // Escape HTML characters and add reset character to close all tags at the end.
+    text = escapeHtmlChars(text) + '\x0f';
 
-    // force there to always be a RESET code at the end of the text to make sure
-    // we close up all open style tags at the end.
-    if (text.length > 0 && text[text.length - 1] !== RESET_CODE) {
-        text = text + RESET_CODE;
-    }
+    // Replace all mIRC formatting characters.
+    // The color character can have arguments.
+    const colorRegex = /[\x02\x1d\x1f\x0f\x16]|\x03(\d{0,2})(?:,(\d{1,2}))?/g;
```

Comment:
You're going to need to do a lot more to convince me that this regexp works. Tell me:
 - What it matches (ideally with an example)
 - What groups does it extract

If this is from another piece of code (after all, we aren't the first people to want to do this), then please link to documentation/source to aid verification.